### PR TITLE
add skipping prefix logic

### PR DESCRIPTION
**Pull request by mukulljangid**
Addresses #278

### CODE REVIEW

File: xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java [hunk 1]
```diff
@@ -152,7 +152,13 @@ public String encodeForSigningClaim(String json) throws JsonProcessingException
    * @return A {@link String} representing the decoded hex.
    */
   public String decode(String hex) {
-    return new BinaryParser(hex).readType(STObjectType.class)
+    String nonSignPrefixHex;
+    if (hex.startsWith(TRX_SIGNATURE_PREFIX) || hex.startsWith(TRX_MULTI_SIGNATURE_PREFIX)) {
+      nonSignPrefixHex = hex.substring(8);
```
Comment (nkramer44) [inline xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java:157]: I know both of these prefixes are 8 characters of hex, but we should probably do `hex.substring(TRX_SIGNATURE_PREFIX.length()` just to be sure, and to make this more readable.

You'll need two `if` blocks to do this fyi

**Review by nkramer44 — CHANGES_REQUESTED** [review]

File: xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java [hunk 2]
```diff
@@ -152,7 +152,15 @@ public String encodeForSigningClaim(String json) throws JsonProcessingException
    * @return A {@link String} representing the decoded hex.
    */
   public String decode(String hex) {
-    return new BinaryParser(hex).readType(STObjectType.class)
+    String nonSignPrefixHex;
+    if (hex.startsWith(TRX_SIGNATURE_PREFIX)) {
+      nonSignPrefixHex = hex.substring(TRX_SIGNATURE_PREFIX.length());
+    } else if (hex.startsWith(TRX_MULTI_SIGNATURE_PREFIX)) {
+      nonSignPrefixHex = hex.substring(TRX_MULTI_SIGNATURE_PREFIX.length());
```
Comment (nkramer44) [inline xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java:159]: let's add a test that hits this

**Review by sappenin — APPROVED** [review]

**Review by sappenin — APPROVED** [review]

File: xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java [hunk 3]
```diff
@@ -122,6 +126,7 @@ public String encodeForMultiSigning(String json, String xrpAccountId) throws Jso
    * @param json String containing JSON to be encoded.
    *
    * @return The binary encoded JSON in hexadecimal form.
+   *
```
Comment (nkramer44) [inline xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java:129]: This formatting seems different than what we have elsewhere. We usually don't have an empty line between `@return` and `@throws`. Not a huge deal but we should double check your intellij settings to reduce PR noise in the future.

File: xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java [hunk 4]
```diff
@@ -145,14 +150,26 @@ public String encodeForSigningClaim(String json) throws JsonProcessingException
   }
 
   /**
-   * Decodes canonical XRPL binary hex string to JSON.
+   * Decodes canonical XRPL binary encodedTransaction string to JSON.
```
Comment (nkramer44) [inline xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java:153]:
```suggestion
   * Decodes canonical XRPL binary hex encoded transaction string to JSON.
```

**Review by nkramer44 — APPROVED** [review]

File: xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java (hunk 3 above)
Comment (mukulljangid) [reply xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java:129]: It was part of my formatter, but I forgot to reformat code before pushing. Thanks for catching this.

**Pull request merged**